Source/GRChomboCore/BoundaryConditions.cpp

```diff
@@ -256,19 +259,93 @@ void BoundaryConditions::fill_reflective_cell(FArrayBox &rhs_box,
+    {
+        // current radius
+        double radius = Coordinates<double>::get_radius(
+            iv, m_dx, {m_center[0], m_center[1], m_center[2]});
+
+        // vector of 2 nearest values and radii within the grid
+        std::array<double, 2> value_at_point;
+        std::array<double, 2> r_at_point;
+        // how many units are we from domain boundary?
+        int units_from_edge = 0;
+        if (a_side == Side::Hi)
+        {
+            // how many units are we from domain boundary?
+            units_from_edge = iv[dir] - m_domain_box.bigEnd(dir);
+            // vector of 2 nearest values and radii within the grid
+            for (int i = 0; i < 2; i++)
+            {
+                IntVect iv_tmp = iv;
+                iv_tmp[dir] += -units_from_edge - i;
+                value_at_point[i] = rhs_box(iv_tmp, icomp);
+                r_at_point[i] = Coordinates<double>::get_radius(
+                    iv_tmp, m_dx, {m_center[0], m_center[1], m_center[2]});
+            }
+        }
+        else // Lo side
+        {
+            // how many units are we from domain boundary?
+            units_from_edge = -iv[dir] + m_domain_box.smallEnd(dir);
+            // vector of 2 nearest values within the grid
+            for (int i = 0; i < 2; i++)
+            {
+                IntVect iv_tmp = iv;
+                iv_tmp[dir] += units_from_edge + i;
+                value_at_point[i] = rhs_box(iv_tmp, icomp);
+                r_at_point[i] = Coordinates<double>::get_radius(
+                    iv_tmp, m_dx, {m_center[0], m_center[1], m_center[2]});
+            }
+        }
+
+        // assume some radial dependence and fit it
+        double analytic_change = 0.0;
+        // comp = const
+        if (order == 0)
+        {
+            analytic_change = 0.0;
+        }
+        // comp = B + A*r
+        else if (order == 1)
+        {
+            double delta_r_in_domain = r_at_point[1] - r_at_point[0];
+            double A =
+                (value_at_point[1] - value_at_point[0]) / delta_r_in_domain;
+            double delta_r_here = radius - r_at_point[0];
+            analytic_change = A * delta_r_here;
+        }
+        // other orders not supported yet
+        else
+        {
+            MayDay::Error("Order not supported for boundary extrapolation.");
+        }
+
+        // set the value here to the extrapolated value
+        rhs_box(iv, icomp) = value_at_point[0] + analytic_change;
+    }
+}
+
 /// Fill the boundary values appropriately based on the params set
 /// in the direction dir
 void BoundaryConditions::fill_boundary_rhs_dir(const Side::LoHiSide a_side,
                                                const GRLevelData &a_soln,
                                                GRLevelData &a_rhs,
-                                               const int dir)
+                                               const int dir,
+                                               const bool filling_rhs)
```

Comment:
Does this argument do anything?

Reply:
Good point! Here it does not, but I do use it in another case where I use mixed boundary conditions (ie, some vars are sommerfeld, some extrapolating). In that case different things happen depending on whether we are filling the rhs or not - e.g. we ignore the sommerfeld vars when not doing rhs. So I thought it might be a useful addition in the longer term.

Reply:
It did make me wonder if we should rename the function so it is less fudgy.